extendeddatacrossword_test.go

```diff
@@ -20,139 +20,150 @@ type PseudoFraudProof struct {
+			colRoots := original.ColRoots()
+
+			t.Run("MaximumErasures", func(t *testing.T) {
+				flattened := original.flattened()
+				flattened[0], flattened[2], flattened[3] = nil, nil, nil
+				flattened[4], flattened[5], flattened[6], flattened[7] = nil, nil, nil, nil
+				flattened[8], flattened[9], flattened[10] = nil, nil, nil
+				flattened[12], flattened[13] = nil, nil
+
+				// Re-import the data square.
+				eds, err := ImportExtendedDataSquare(flattened, codec, NewDefaultTree)
+				if err != nil {
+					t.Errorf("ImportExtendedDataSquare failed: %v", err)
+				}
+
+				err = eds.Repair(rowRoots, colRoots)
+				if err != nil {
+					t.Errorf("unexpected err while repairing data square: %v, codec: :%s", err, codecName)
+				} else {
+					assert.Equal(t, original.GetCell(0, 0), ones)
+					assert.Equal(t, original.GetCell(0, 1), twos)
+					assert.Equal(t, original.GetCell(1, 0), threes)
+					assert.Equal(t, original.GetCell(1, 1), fours)
+				}
+			})
+			t.Run("Unrepairable", func(t *testing.T) {
+				flattened := original.flattened()
+				flattened[0], flattened[2], flattened[3] = nil, nil, nil
+				flattened[4], flattened[5], flattened[6], flattened[7] = nil, nil, nil, nil
+				flattened[8], flattened[9], flattened[10] = nil, nil, nil
+				flattened[12], flattened[13], flattened[14] = nil, nil, nil
+
+				// Re-import the data square.
+				eds, err := ImportExtendedDataSquare(flattened, codec, NewDefaultTree)
+				if err != nil {
+					t.Errorf("ImportExtendedDataSquare failed: %v", err)
+				}
+
+				err = eds.Repair(rowRoots, colRoots)
+				if err == nil {
+					t.Errorf("did not return an error on trying to repair an unrepairable square")
+				}
+			})
+			t.Run("CorruptedEDS", func(t *testing.T) {
+				var corrupted ExtendedDataSquare
+				var byzData *ErrByzantineData
+				corrupted, err = original.deepCopy(codec)
+				if err != nil {
+					t.Fatalf("unexpected err while copying original data: %v, codec: :%s", err, codecName)
+				}
+				corruptChunk := bytes.Repeat([]byte{66}, bufferSize)
+				corrupted.setCell(0, 0, corruptChunk)
+				err = corrupted.Repair(rowRoots, colRoots)
+				if err == nil {
+					t.Errorf("did not return an error on trying to repair a square with bad roots")
+				}
+				t.Run("BadRowInOriginalData", func(t *testing.T) {
+					corrupted, err = original.deepCopy(codec)
+					if err != nil {
+						t.Fatalf("unexpected err while copying original data: %v, codec: :%s", err, codecName)
+					}
+					corrupted.setCell(0, 0, corruptChunk)
+					err = corrupted.Repair(corrupted.getRowRoots(), corrupted.getColRoots())
+					if !errors.As(err, &byzData) || byzData.Axis != Row {
+						t.Errorf("did not return a ErrByzantineData for a bad row; got: %v", err)
+					}
+					// Construct the fraud proof
+					fraudProof := PseudoFraudProof{0, byzData.Index, byzData.Shares}
+					// Verify the fraud proof
+					// TODO in a real fraud proof, also verify Merkle proof for each non-nil share.
```

Comment:
Should an issue be created out of this TODO?

Reply:
Hmmm. It's not really a TODO as much as it is a note. Constructing the Merkle proofs would only be possible for a subset of non-`nil` shares, and verifying them would be a lot of extra work that isn't directly related to rsmt2d.